pkg/protocols/common/hosterrorscache/hosterrorscache.go

```diff
@@ -75,24 +75,34 @@ func (c *Cache) Close() {
 	c.failedTargets.Purge()
 }
 
-func (c *Cache) normalizeCacheValue(value string) string {
-	finalValue := value
-	if strings.HasPrefix(value, "http") {
-		if parsed, err := url.Parse(value); err == nil {
-			hostname := parsed.Host
-			finalPort := parsed.Port()
-			if finalPort == "" {
-				if parsed.Scheme == "https" {
-					finalPort = "443"
-				} else {
-					finalPort = "80"
-				}
-				hostname = net.JoinHostPort(parsed.Host, finalPort)
+// NormalizeCacheValue processes the input value and returns a normalized cache
+// value.
+func (c *Cache) NormalizeCacheValue(value string) string {
+	var normalizedValue string = value
+
+	u, err := url.ParseRequestURI(value)
+	if err != nil || u.Host == "" {
+		u, err2 := url.ParseRequestURI("https://" + value)
+		if err2 != nil {
+			return normalizedValue
+		}
+
+		normalizedValue = u.Host
```

Comment:
we can use urlutil.ParseAbsoluteURL() which handles internal normalization and strict parsing of urls its in `projectdiscovery/utils/url`

Reply:
The idea for using `url.ParseRequestURI` instead of `url.Parse` was to avoid being too strict. This way, the input can accept `host:port` rather than requiring full `scheme://host:port`.